athina/evals/function/functions.py

```diff
@@ -60,7 +238,92 @@ def contains_json(response=None):
             "reason": "Output does not contain JSON",
         }
 
-def no_invalid_links(response=None):
+def is_email(response):
+    """
+    Check if the response is a valid email address.
+
+    Args:
+        response (str): The response string to check for a valid email address.
+
+    Returns:
+        dict: A dictionary containing the result of the email address check and the reason for the result.
+    """
+    return regex(r'^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$', response)
+
+def is_phone_number(response):
+    """
+    Check if the response is a valid phone number.
+
+    Args:
+        response (str): The response string to check for a valid phone number.
+
+    Returns:
+        dict: A dictionary containing the result of the phone number check and the reason for the result.
+    """
+    return regex(r'^\+?1?\d{9,15}$', response)
+
+def contains_link(response):
+    """
+    Check if the response contains a link.
+
+    Args:
+        response (str): The response string to check for a link.
+
+    Returns:
+        dict: A dictionary containing the result of the link check and the reason for the result.
+    """
+    pattern = r"(?!.*@)(?:https?://)?(?:www\.)?\S+\.\S+"
+    result = bool(re.search(pattern, response))
+    if result:
+        return {"result": True, "reason": "Link found in output"}
+    else:
+        return {"result": False, "reason": "No link found in output"}
+
+def contains_valid_link(response):
+    """
+    Check if the response contains a valid link.
+
+    Args:
+        response (str): The response string to check for a valid link.
+
+    Returns:
+        dict: A dictionary containing the result of the link check and the reason for the result.
+    """
+    pattern = r"(?!.*@)(?:https?://)?(?:www\.)?\S+\.\S+"
+    link_match = re.search(pattern=pattern, string=response)
+    if link_match:
+        matched_url = link_match.group()
+        if matched_url:
+            standardized_url = _standardize_url(matched_url)
+            try:
+                response = requests.head(standardized_url)
+                if response.status_code == 200:
+                    return {
+                        "result": True,
+                        "reason": f"link {matched_url} found in output and is valid",
+                    }
+                else:
+                    return {
+                        "result": False,
+                        "reason": f"link {matched_url} found in output but is invalid",
+                    }
+            except:
```

Comment:
I think we could handle the exceptions better here.

Should probably have different `reason` messages for 401, 404, 500.